File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/estoque/RequisicaoInternaCabecalho.java
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [REQUISICAO_INTERNA_CABECALHO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.estoque;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.util.Date;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.t2ti.fenix.model.cadastros.Colaborador;

import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.util.Set;
import javax.persistence.OneToMany;
import javax.persistence.CascadeType;

@Entity
@Table(name="REQUISICAO_INTERNA_CABECALHO")
@NamedQuery(name="RequisicaoInternaCabecalho.findAll", query="SELECT t FROM RequisicaoInternaCabecalho t")
public class RequisicaoInternaCabecalho implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_REQUISICAO")
	private Date dataRequisicao;

    @Column(name="SITUACAO")
	private String situacao;

    @ManyToOne
	@JoinColumn(name="ID_COLABORADOR")
    private Colaborador colaborador;

    @OneToMany(mappedBy = "requisicaoInternaCabecalho", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<RequisicaoInternaDetalhe> listaRequisicaoInternaDetalhe;

	public RequisicaoInternaCabecalho() {
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public Date getDataRequisicao() {
		return this.dataRequisicao;
	}

	public void setDataRequisicao(Date dataRequisicao) {
		this.dataRequisicao = dataRequisicao;
	}

    public String getSituacao() {
		return this.situacao;
	}

	public void setSituacao(String situacao) {
		this.situacao = situacao;
	}

    public Colaborador getColaborador() {
		return this.colaborador;
	}

	public void setColaborador(Colaborador colaborador) {
		this.colaborador = colaborador;
	}

    public Set<RequisicaoInternaDetalhe> getListaRequisicaoInternaDetalhe() {
		return this.listaRequisicaoInternaDetalhe;
	}

	public void setListaRequisicaoInternaDetalhe(Set<RequisicaoInternaDetalhe> listaRequisicaoInternaDetalhe) {
		this.listaRequisicaoInternaDetalhe = listaRequisicaoInternaDetalhe;
		for (RequisicaoInternaDetalhe requisicaoInternaDetalhe : listaRequisicaoInternaDetalhe) {
			requisicaoInternaDetalhe.setRequisicaoInternaCabecalho(this);
		}
	}

		
}

File: fontes/frontend/fenix_provider/lib/src/model/estoque/requisicao_interna_cabecalho.dart
/*
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [REQUISICAO_INTERNA_CABECALHO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';

import 'package:intl/intl.dart';
import 'package:fenix/src/model/model.dart';

class RequisicaoInternaCabecalho {
	int? id;
	int? idColaborador;
	DateTime? dataRequisicao;
	String? situacao;
	Colaborador? colaborador;
	List<RequisicaoInternaDetalhe>? listaRequisicaoInternaDetalhe = [];

	RequisicaoInternaCabecalho({
			id,
			idColaborador,
			dataRequisicao,
			situacao,
			colaborador,
			listaRequisicaoInternaDetalhe,
		});

	static List<String> campos = <String>[
		'ID', 
		'DATA_REQUISICAO', 
		'SITUACAO', 
	];
	
	static List<String> colunas = <String>[
		'Id', 
		'Data da Requisição', 
		'Situação da Requisição', 
	];

	RequisicaoInternaCabecalho.fromJson(Map<String, dynamic> jsonDados) {
		id = jsonDados['id'];
		idColaborador = jsonDados['idColaborador'];
		dataRequisicao = jsonDados['dataRequisicao'] != null ? DateTime.tryParse(jsonDados['dataRequisicao']) : null;
		situacao = getSituacao(jsonDados['situacao']);
		colaborador = jsonDados['colaborador'] == null ? null : Colaborador.fromJson(jsonDados['colaborador']);
		listaRequisicaoInternaDetalhe = (jsonDados['listaRequisicaoInternaDetalhe'] as Iterable?)?.map((m) => RequisicaoInternaDetalhe.fromJson(m)).toList() ?? [];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonDados = <String, dynamic>{};

		jsonDados['id'] = id ?? 0;
		jsonDados['idColaborador'] = idColaborador ?? 0;
		jsonDados['dataRequisicao'] = dataRequisicao != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataRequisicao!) : null;
		jsonDados['situacao'] = setSituacao(situacao);
		jsonDados['colaborador'] = colaborador == null ? null : colaborador!.toJson;
		

		var listaRequisicaoInternaDetalheLocal = [];
		for (RequisicaoInternaDetalhe objeto in listaRequisicaoInternaDetalhe ?? []) {
			listaRequisicaoInternaDetalheLocal.add(objeto.toJson);
		}
		jsonDados['listaRequisicaoInternaDetalhe'] = listaRequisicaoInternaDetalheLocal;
	
		return jsonDados;
	}
	
    getSituacao(String? situacao) {
    	switch (situacao) {
    		case 'A':
    			return 'Aberta';
    		case 'D':
    			return 'Deferida';
    		case 'I':
    			return 'Indeferida';
    		default:
    			return null;
    		}
    	}

    setSituacao(String? situacao) {
    	switch (situacao) {
    		case 'Aberta':
    			return 'A';
    		case 'Deferida':
    			return 'D';
    		case 'Indeferida':
    			return 'I';
    		default:
    			return null;
    		}
    	}


	String objetoEncodeJson(RequisicaoInternaCabecalho objeto) {
	  final jsonDados = objeto.toJson;
	  return json.encode(jsonDados);
	}
	
}